spec/requests/groups_controller_spec.rb

```diff
@@ -858,6 +858,54 @@ def expect_type_to_return_right_groups(type, expected_group_ids)
 
       context "user default notifications" do
         it "should update default notification preference for existing users" do
+          user1 = Fabricate(:user)
+          user2 = Fabricate(:user)
+          group.add(user1)
+          group.add(user2)
+
+          put "/groups/#{group.id}.json", params: {
+            group: {
+              default_notification_level: 2
+            }
+          }
+
+          expect(response.status).to eq(200)
+          group_users = group.group_users
+          expect(response.parsed_body["user_count"]).to eq(group_users.count)
+
+          group_users.where(user: user1).update_all(notification_level: 1)
+
+          put "/groups/#{group.id}.json", params: {
+            group: {
+              default_notification_level: 2
+            }
+          }
+
+          expect(response.status).to eq(200)
+          expect(response.parsed_body["user_count"]).to eq(group.group_users.count - 1)
+
+          put "/groups/#{group.id}.json", params: {
+            group: {
+              default_notification_level: 2
+            },
+            update_existing_users: true
+          }
+
+          expect(response.status).to eq(200)
+          expect(response.parsed_body["success"]).to eq("OK")
+          expect(GroupUser.exists?(user: user2, group: group, notification_level: 2)).to be_truthy
+
+          put "/groups/#{group.id}.json", params: {
+            group: {
+              default_notification_level: 2
+            }
+          }
+
+          expect(response.status).to eq(200)
+          expect(response.parsed_body["success"]).to eq("OK")
```

Comment:
I'm wondering if we need this final assertion since it doesn't test for any state change on the server side. The `PUT` request here should be idempotent so there is no need for us to test for a 200 response.

Reply:
Here it's correctly returning `success: true` instead of `user_count: 3` like in [this line](https://github.com/discourse/discourse/pull/14084/files#diff-1f3b991714a19291494dd9505a88dd6602f33c13de7a1eea55b664489aebac9eR866-R874). This final assertion is little different from the first one.

Reply:
Do you mean `success: 'OK'` is not always in the response even if the status code is 200?

Reply:
yes, it will return `user_count: X` in some conditions.